tests/main/lxd-postrm-purge/prep-snapd-in-lxd.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/sh
+
+set -ex
+
+# XXX: remove once the "umount /snap" change in postrm has propagated all
+#      the way to the image
+if [ -e /var/lib/dpkg/info/snapd.postrm ]; then
+    # ensure we can umount /snap
+    sed -i 's#echo "Final directory cleanup"#umount /snap || true#' /var/lib/dpkg/info/snapd.postrm
+fi
+
+# wait for cloud-init to finish before doing any apt operations, since it will
+# re-write the apt sources.list file and we will be racing with the re-write 
+# trying to do apt operations before cloud-init is done
+# TODO: we should eventually use `cloud-init status --wait`, but that doesn't work
+# in nested containers, see https://bugs.launchpad.net/cloud-init/+bug/1905493
+for _ in $(seq 1 60); do
+    if python3 -c "import apt;apt.apt_pkg.SourceList().read_main_list()"; then
```

Comment:
Is this sufficient? I'm wondering what particular property is this code checking. That apt lock is released? Is cloud init holding the lock for long period of time without releasing it? Is there a chance that while cloud-init is still working, it is temporarily releasing the lock so that this code can succeed, but apt operations in cloud-init are not yet fully done?

As an alternative, is there any file that cloud init writes at the end of it's job that we could monitor for instead?

Reply:
This checks that we have a valid sources.list. Cloud-init uses an invalid sources.list template initially that will not be parsed by "read_main_list".

Reply:
Ah, that makes sense. Perhaps add a comment like that in another pass.

Reply:
As the TODO above this check says, there is a bug in cloud-init that makes it never finish waiting (actually due to cloud-init blocking on snapd which will never succeed in nested lxd containers), and thus there is no file that would indicate that cloud-init is "fully done" inside the nested lxd container and so we have resorted to this hack which waits for cloud-init to at least have progressed far enough such that the /etc/apt/sources.list file is valid.